plugins/store/src/store.rs

```diff
@@ -174,142 +164,206 @@ impl<R: Runtime> StoreBuilder<R> {
+    pub fn disable_auto_save(mut self) -> Self {
+        self.auto_save = None;
+        self
+    }
+
+    pub(crate) fn build_inner(
+        mut self,
+        mut stores: MutexGuard<'_, HashMap<PathBuf, ResourceId>>,
+    ) -> crate::Result<(Arc<Store<R>>, ResourceId)> {
+        if stores.contains_key(&self.path) {
+            return Err(crate::Error::AlreadyExists(self.path));
+        }
+
+        let mut store_inner = StoreInner::new(
+            self.app.clone(),
+            self.path.clone(),
+            self.defaults.take(),
+            self.serialize_fn,
+            self.deserialize_fn,
+        );
+        let _ = store_inner.load();
+
+        let store = Store {
+            auto_save: self.auto_save,
+            auto_save_debounce_sender: Arc::new(Mutex::new(None)),
+            store: Arc::new(Mutex::new(store_inner)),
+        };
+
+        let store = Arc::new(store);
+        let rid = self.app.resources_table().add_arc(store.clone());
+        stores.insert(self.path, rid);
+
+        Ok((store, rid))
+    }
+
+    /// Builds the [`Store`], also see [`build_or_existing`](Self::build_or_existing).
+    ///
+    /// This loads the store from disk and put the store in the app's resource table,
+    /// to remove it from the resource table, call [`Store::close_store`]
+    ///
+    /// # Errors
+    ///
+    /// If a store with this path is already in the resource table,
+    /// will return a [`crate::Error::AlreadyExists`]
     ///
     /// # Examples
     /// ```
     /// tauri::Builder::default()
     ///   .plugin(tauri_plugin_store::Builder::default().build())
     ///   .setup(|app| {
-    ///     let store = tauri_plugin_store::StoreBuilder::new(app, "store.json").build();
+    ///     let store = tauri_plugin_store::StoreBuilder::new(app, "store.json").new_or_existing()?;
     ///     Ok(())
     ///   });
     /// ```
-    pub fn build(self) -> Store<R> {
-        let collection = self.app.state::<StoreCollection<R>>();
-        let mut stores = collection.stores.lock().unwrap();
-        let store = stores
-            .get(&self.path)
-            .and_then(|store| store.upgrade())
-            .unwrap_or_else(|| {
-                let mut store = StoreInner::new(self.app.clone(), self.path.clone());
-                let _ = store.load(self.deserialize);
-                let store = Arc::new(Mutex::new(store));
-                stores.insert(
-                    self.path.clone(),
-                    Arc::<Mutex<StoreInner<R>>>::downgrade(&store),
-                );
-                store
-            });
-        drop(stores);
-        Store {
-            defaults: self.defaults,
-            serialize: self.serialize,
-            deserialize: self.deserialize,
-            auto_save: self.auto_save,
-            auto_save_debounce_sender: Arc::new(Mutex::new(None)),
-            store,
+    pub fn create(self) -> crate::Result<Arc<Store<R>>> {
```

Comment:
I don't quite like `create` on a builder, I think it should be kept as `build`

Reply:
+1 for `build` and `build_or_existing` @lucasfernog